### PR TITLE
feat: add pagination to trash page 

### LINKED ISSUE #1493: Paginate recycle bin

### PR DESCRIPTION
**Pull request by pyphilia**
Add pagination to recycled items. Remove filter and ordering options for simplification.

close #1493

### CODE REVIEW

**Review by ReidyT — CHANGES_REQUESTED** [review]
Thanks for this PR, it looks good to me! I just let you some comments and suggestions, I'm of course open to the discussion 🫠

File: src/components/pages/RecycledItemsScreen.tsx [hunk 1]
```diff
@@ -26,86 +24,63 @@ import {
...
-    ?.filter(
-      (d) =>
-        shouldDisplayItem(d.type) &&
-        d.name.toLowerCase().includes(searchText.toLocaleLowerCase()),
-    )
-    ?.sort(sortFn);
+
+  const { data, fetchNextPage, isLoading, isFetching } =
+    hooks.useInfiniteOwnRecycledItemData(
+      // todo: adapt page size given the user window height
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply src/components/pages/RecycledItemsScreen.tsx:37]: The only "problem" currently is that if you have a big screen, you will see only 10 items, and then will have to click load. It's still okay, but not so efficient.

[the remark this answers is not in the document]
Comment (ReidyT) [reply src/components/pages/RecycledItemsScreen.tsx:37]: Yeah I can understand that. I initially thought it was a pagination where user have to click on next page or on the page number to navigate. But in the case of a button "load more" or an infinite scroll, I think it's okay to have less or more items depending on the screen.

File: src/components/pages/RecycledItemsScreen.tsx [hunk 2]
```diff
@@ -45,16 +45,13 @@ const RecycledItemsScreenContent = (): JSX.Element => {
...
+        .flatMap((p) => p.data)
         ?.map((p) => p.item);
 
-      const totalFetchedItems = data
-        ? data.pages.map(({ data: d }) => d.length).reduce((a, b) => a + b, 0)
-        : 0;
+      const totalFetchedItems = fetchedItems.length;
 
-      const hasSelection = Boolean(selectedIds.length && filteredData?.length);
+      const hasSelection = Boolean(selectedIds.length && fetchedItems?.length);
```
Comment (ReidyT) [inline src/components/pages/RecycledItemsScreen.tsx:54]: Still needed? 
```suggestion
      const hasSelection = Boolean(selectedIds.length && fetchedItems.length);
```

**Review by ReidyT — APPROVED** [review]
Thanks for your changes, LGTM ⛳️

**Pull request merged**